### ISSUE TITLE
Colors reversed from typically-expected meanings in Variable Explorer "background color" mode, leading to cognitive dissonance

### ISSUE THREAD

**Issue by CAM-Gerlach** [spyder-ide/spyder#6994]
## Problem Description

At least where I'm from (the US), warm colors (reds, etc) are almost universally used to mean high values (e.g. temperature) when used in contrast with cool colors (blues) to mean low values, and this is in fact where the term "heatmap" comes from, the "warmer" areas being those with more of a certain quantity (which is essentially what the colors are supposed to show in Spyder). However, for some reason, the colors are "backward" from this standard convention, leading to constant cognitive dissonance when evaluating the data using the otherwise useful pseudo-heatmaps, and there is no means to reverse it. 

Unless this is common in many parts of the world, which I not been able to find much evidence of in my limited searches, then the default should be reversed for sanity's sake. Opinions/feedback are, of course, welcome before we proceed.

## Package Versions

* Spyder version: 3.2.8 
* Python version: 3.6.4
* Qt version: 5.6.2
* PyQt5 version: 5.6
* Operating system: Windows 8.1

### Dependencies

```
pyflakes >=0.6.0 :  1.6.0 (OK)
pycodestyle >=2.3:  2.3.1 (OK)
pygments >=2.0   :  2.2.0 (OK)
pandas >=0.13.1  :  0.22.0 (OK)
numpy >=1.7      :  1.14.0 (OK)
sphinx >=0.6.6   :  1.7.2 (OK)
rope >=0.9.4     :  0.10.7 (OK)
jedi >=0.9.0     :  0.11.1 (OK)
nbconvert >=4.0  :  5.3.1 (OK)
sympy >=0.7.3    :  1.1.1 (OK)
cython >=0.21    :  0.27.3 (OK)
qtconsole >=4.2.0:  4.3.1 (OK)
IPython >=4.0    :  6.2.1 (OK)
pylint >=0.25    :  1.8.2 (OK)
```

**Comment by jnsebgosselin** [comment]
I agree with this. 

I'm wondering also if we should use a sequential colormap instead of diverging one... I think this may be better.
https://matplotlib.org/2.0.2/examples/color/colormaps_reference.html

Below, a screenshot of the current situation.
![image](https://user-images.githubusercontent.com/10170372/41111218-5dbdcb12-6a49-11e8-9ce9-97019ee2094d.png)

**Comment by CAM-Gerlach** [comment]
> I'm wondering also if we should use a sequential colormap instead of diverging one...

Actually, I think a diverging one makes more sense here. Both maxima and minima may be important for most distributions (i.e. mean Guassian, the most reasonable rough approximation for the real world), as a diverging colormap emphasizes both, while making the middle values (i.e. likely the majority) more neutral, and is overall less gaudy. The sequential ones, meanwhile, would be better if we knew a distribution was heavily right-tailed, but generally that's a much less safe assumption than Gaussian.

**Comment by jnsebgosselin** [comment]
> Actually, I think a diverging one makes more sense here. Both maxima and minima may be important for most distributions (i.e. mean Guassian, the most reasonable rough approximation for the real world), as a diverging colormap emphasizes both, while making the middle values (i.e. likely the majority) more neutral, and is overall less gaudy. The sequential ones, meanwhile, would be better if we knew a distribution was heavily right-tailed, but generally that's a much less safe assumption than Gaussian.

It is a good point, but I would say that it depends what your dataset represents and what you are looking for.  I often prefer to use a sequential colormaps to represent the kind of data I'm working with, but as I said, it depends.

An option would be to give some choices to the user regarding the colormap they want to use. Something like adding a combobox with different colormap would be great.

**Comment by jnsebgosselin** [comment]
We could replace the `Background color` checkbox by a combobox containing options for the colormap. The first item of the combobox should be `None`. 

![image](https://user-images.githubusercontent.com/10170372/41117976-81570652-6a5c-11e8-8e65-d29b5b38b680.png)

**Comment by CAM-Gerlach** [comment]
That would indeed be a nice feature to have, although we wouldn't be able to fit very many there. However, IMO that goes well beyond the scope of this PR, which is really just a simple fix for the color direction being reversed, and might be better addressed in a follow-up.

**Comment by jnsebgosselin** [comment]
> That would indeed be a nice feature to have, although we wouldn't be able to fit very many there.

We don't need too. Only a few choices would be necessary imo.

**Comment by CAM-Gerlach** [comment]
Yeah; we could have None, Sequential, Diverging, and maybe (Perceptually) Uniform, although even just the first three would take up somewhat more space than the existing ``Background Color`` checkbox, limiting the minimum window size for dataframes.

In any case, sounds like a good idea; its just out of scope of this one, which is to fix an issue with the existing color table rather than adding new ones.

**Comment by jnsebgosselin** [comment]
> its just out of scope of this one, which is to fix an issue with the existing color table rather than adding new ones.

I agree. I took the liberty to brainstorm a little bit around the idea since this is an issue and not PR. I'll open a new Issue to propose the idea of adding more colormap choices to the variable explorer.

> [...] although even just the first three would take up somewhat more space than the existing Background Color checkbox, limiting the minimum window size for dataframes.

If we add this in a combobox, there is no reason for this to take up more space than the actual checkbox. But like I said, let's discussed this a a new issue.

**Comment by CAM-Gerlach** [comment]
> I agree. I took the liberty to brainstorm a little bit around the idea since this is an issue and not PR.

No worries! Good ideas.

> If we add this in a combobox, there is no reason for this to take up more space than the actual checkbox.

You're totally right, silly me! I was still thinking of radio buttons for some reason. I guess the downside of that would be that its much less quick to toggle on/off—maybe a combobox with the colormaps and a simple checkbox to its left to enable/disable it? But like you said, we can discuss in another issue.
